src/haxelib/api/Installer.hx

```diff
@@ -0,0 +1,852 @@
+		final path = FileSystem.fullPath(path);
+		userInterface.log('Installing $path');
+		// read zip content
+		final zip = FsUtils.unzip(path);
+
+		final info = Data.readInfos(zip, false);
+		final library = info.name;
+		final version = info.version;
+
+		installZip(info.name, info.version, zip);
+
+		// set current version
+		scope.setVersion(library, version);
+
+		userInterface.log('  Current version is now $version');
+		userInterface.log("Done");
+
+		handleDependencies(library, version, info.dependencies);
+	}
+
+	/**
+		Clears memory on git or hg library branches.
+
+		An installer instance keeps track of updated vcs dependencies
+		to avoid cloning the same branch twice.
+
+		This function can be used to clear that memory.
+	**/
+	public function forgetVcsBranches():Void {
+		vcsBranchesByLibraryName.clear();
+	}
+
+	/** Installs libraries from the `haxelib.json` file at `path`. **/
+	public function installFromHaxelibJson(path:String) {
+		final path = FileSystem.fullPath(path);
+		userInterface.log('Installing libraries from $path');
+
+		final dependencies = Data.readData(File.getContent(path), false).dependencies;
+
+		try
+			installFromDependencies(dependencies)
+		catch (e)
+			throw new InstallationException('Failed installing dependencies from $path:\n$e');
+	}
+
+	/**
+		Installs the libraries required to build the HXML file at `path`.
+
+		Throws an error when trying to install a library from the haxelib
+		server if the library has no versions or if the requested
+		version does not exist.
+
+		If `confirmHxmlInstall` is passed in, it will be called with information
+		about the libraries to be installed, and the installation only proceeds if
+		it returns `true`.
+	 **/
+	public function installFromHxml(path:String, ?confirmHxmlInstall:(libs:Array<{name:ProjectName, version:Version}>) -> Bool) {
+		final path = FileSystem.fullPath(path);
+		userInterface.log('Installing all libraries from $path:');
+		final libsToInstall = LibFlagData.fromHxml(path);
+
+		if (libsToInstall.empty())
+			return;
+
+		// Check the version numbers are all good
+		userInterface.log("Loading info about the required libraries");
+
+		final installData = getFilteredInstallData(libsToInstall);
+
+		final libVersions = [
+			for (library in installData)
+				{name:library.name, version:library.version}
+		];
+		// Abort if not confirmed
+		if (confirmHxmlInstall != null && !confirmHxmlInstall(libVersions))
+			return;
+
+		for (library in installData) {
+			try
+				installFromInstallData(library.name, library.installData)
+			catch (e) {
+				userInterface.log(e.toString());
+				continue;
+			}
+
+			final libraryName = switch library.installData {
+				case VcsInstall(version, vcsData): getName(library.name, version, vcsData.subDir);
+				case _: library.name;
+			}
+
+			if (library.isLatest || !scope.isLibraryInstalled(libraryName))
+				setVersionAndLog(libraryName, library.installData);
+
+			userInterface.log("Done");
```

Comment:
Done what? This doesn't seem to have a partner print in this loop.

Reply:
There are a bunch of these unpaired "Done"s in this file, but I tried to keep the interface as is. Might be better to cleanup messages like this all at once in another PR.